src/webview_provider/insightsPanel.ts

```diff
@@ -0,0 +1,465 @@
+type UpdateConfigPropsArray = {
+  config: UpdateConfigProps[];
+  projectRoot: string;
+};
+export interface DeferConfig {
+  deferToProduction: boolean;
+  favorState: boolean;
+  manifestPathForDeferral: string;
+  manifestPathType?: ManifestPathType;
+  dbtCoreIntegrationId?: number;
+}
+
+interface DbtProject {
+  projectRoot: string;
+  projectName: string;
+}
+
+@provideSingleton(InsightsPanel)
+export class InsightsPanel extends AltimateWebviewProvider {
+  public static readonly viewType = "dbtPowerUser.Insights";
+  protected viewPath = "/insights";
+  protected panelDescription = "Toggle Defer to prod and other features";
+
+  public constructor(
+    protected dbtProjectContainer: DBTProjectContainer,
+    protected altimateRequest: AltimateRequest,
+    protected telemetry: TelemetryService,
+    protected emitterService: SharedStateService,
+    protected dbtTerminal: DBTTerminal,
+    private deferToProdService: DeferToProdService,
+  ) {
+    super(
+      dbtProjectContainer,
+      altimateRequest,
+      telemetry,
+      emitterService,
+      dbtTerminal,
+    );
+
+    this._disposables.push(
+      window.onDidChangeActiveTextEditor(
+        async (event: TextEditor | undefined) => {
+          if (event === undefined) {
+            return;
+          }
+
+          if (this._panel) {
+            const currentProject = await this.getCurrentProject();
+
+            const dbtIntegrationMode = workspace
+              .getConfiguration("dbt")
+              .get<string>("dbtIntegration", "core");
+
+            const projectPath = this.getCurrentProject();
+            if (!projectPath) {
+              throw new Error("No project selected");
+            }
+
+            this._panel!.webview.postMessage({
+              command: "renderDeferConfig",
+              args: {
+                config: this.deferToProdService.getDeferConfigByProjectRoot(
+                  projectPath.projectRoot.fsPath,
+                ),
+                projectPath: currentProject?.projectRoot.fsPath,
+                dbtIntegrationMode,
+              },
+            });
+          }
+        },
+      ),
+    );
+  }
+
+  private getCurrentProject() {
+    const currentFilePath = window.activeTextEditor?.document.uri;
+    if (!currentFilePath) {
+      throw new Error("No file selected in the editor");
+    }
+
+    const currentProject =
+      this.dbtProjectContainer.findDBTProject(currentFilePath);
+    return currentProject;
+  }
+
+  private async updateDeferConfig(
+    syncRequestId: string | undefined,
+    params: UpdateConfigPropsArray,
+  ) {
+    try {
+      this.dbtTerminal.debug("InsightsPanel", "Updating defer config", params);
+      if (!params.projectRoot) {
+        window.showErrorMessage("Please select a project");
+        return;
+      }
+      // defer config is preview feature, then check keys
+      if (!this.altimateRequest.handlePreviewFeatures()) {
+        throw new Error("Invalid credentials");
+      }
```

Comment:
It should be allowed to choose the local manifest file and deactivate the defer to prod feature in case the user doesn't have an altimate key anymore (previously having setup it).